src/Sylius/Component/Shipping/spec/Checker/Eligibility/CategoryRequirementEligibilityCheckerSpec.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace spec\Sylius\Component\Shipping\Checker\Eligibility;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use PhpSpec\ObjectBehavior;
+use Sylius\Component\Shipping\Checker\Eligibility\ShippingMethodEligibilityCheckerInterface;
+use Sylius\Component\Shipping\Model\ShippableInterface;
+use Sylius\Component\Shipping\Model\ShippingCategoryInterface;
+use Sylius\Component\Shipping\Model\ShippingMethodInterface;
+use Sylius\Component\Shipping\Model\ShippingSubjectInterface;
+
+final class CategoryRequirementEligibilityCheckerSpec extends ObjectBehavior
+{
+    function it_implements_shipping_method_eligibility_checker_interface(): void
+    {
+        $this->shouldImplement(ShippingMethodEligibilityCheckerInterface::class);
+    }
+
+    function it_approves_category_requirement_if_categories_match(
+        ShippingSubjectInterface $subject,
+        ShippingMethodInterface $shippingMethod,
+        ShippingCategoryInterface $shippingCategory,
+        ShippableInterface $shippable
+    ): void {
+        $shippingMethod->getCategory()->willReturn($shippingCategory);
+        $shippingMethod->getCategoryRequirement()->willReturn(ShippingMethodInterface::CATEGORY_REQUIREMENT_MATCH_ANY);
+
+        $shippable->getShippingCategory()->willReturn($shippingCategory);
+        $subject->getShippables()->willReturn(new ArrayCollection([$shippable->getWrappedObject()]));
+
+        $this->isEligible($subject, $shippingMethod)->shouldReturn(true);
```

Comment:
I think we should cover all possible approval cases. We can either write a specification for each of possible result or pack them into approde/decline groups or per type (any, none, all). If you would like to group them, here is a useful sample of repeated usage of the same config in phpspec:
```suggestion
        $shippingMethod->getCategory()->willReturn($shippingCategory);
        $shippingMethod->getCategoryRequirement()->willReturn(
            ShippingMethodInterface::CATEGORY_REQUIREMENT_MATCH_ANY,
            ShippingMethodInterface::CATEGORY_REQUIREMENT_MATCH_ALL,
            ShippingMethodInterface::CATEGORY_REQUIREMENT_MATCH_NONE
        ); # every time we will execute this test different requirement will be returned

        $shippable->getShippingCategory()->willReturn(
            $shippingCategory,
            $shippingCategory,
            $shippingCategory2
        ); # first two times we will return the same shippingCategory. But we need different for the third test to pass
        
        $subject->getShippables()->willReturn(new ArrayCollection([$shippable->getWrappedObject()]));

        $this->isEligible($subject, $shippingMethod)->shouldReturn(true);
        $this->isEligible($subject, $shippingMethod)->shouldReturn(false);
        $this->isEligible($subject, $shippingMethod)->shouldReturn(true);
```

Reply:
Added :)